done

DanilNaum opened the pull request.

DanilNaum committed:
done

Perederey commented on client/internal/ui/forms.go:
```diff
@@ -0,0 +1,182 @@
+
+	form.AddButton("download", func() {
+		progress, errChan := a.serverHandlers.DownloadFile(record.ServerID, path)
+		LOOP:for {
```
В downloadFileForm слишком много логики обработки прогресса смешано с UI. Вынеси логику загрузки в отдельную функцию.

Perederey commented on client/pkg/client/files.go:
```diff
@@ -0,0 +1,216 @@
+
+	// Send file chunks
+	buffer := make([]byte, ChunkSize)
+	LOOP:for {
```
Лучше подход с defer и каналом отмены:

```go
func (c *Client) DownloadFile(recordID, savePath string) (chan int, chan error) {
	progress := make(chan int, 1)
	errChan := make(chan error, 1)
	
	go func() {
		defer close(progress)
		defer close(errChan)
		
		// ... setup code ...
		
		for {
			resp, err := stream.Recv()
			if errors.Is(err, io.EOF) {
				break
			}
			if err != nil {
				errChan <- err
				return
			}
			// ... processing logic ...
		}
	}()
	
	return progress, errChan
}
```

Perederey commented on server/internal/transport/grpc/record/v1/mapper.go:
```diff
@@ -0,0 +1,85 @@
+			},
+		}
+	case entity.RecordTypeText:
+		record.Content = &pb.Record_TextData{
```
Добавь коммент для FileData, объясняющий почему content не устанавливается

Perederey commented on server/internal/transport/grpc/record/v1/handler.go:
```diff
@@ -0,0 +1,273 @@
+type grpcRecordHandler struct {
+	pb.UnimplementedRecordServiceServer
+	jwtManager JWTManager[int]
+	usecase    usecase
```
usecase содержит слишком много методов - раздели его на более мелкие интерфейсы по принципу Interface Segregation

Perederey commented on client/cmd/main.go:
```diff
@@ -0,0 +1,34 @@
+	"github.com/rivo/tview"
+)
+
+func main() {
```
Слишком много инициализации в функции main - вынеси создание зависимостей в отдельную функцию или попробуй еще uber Fx

Perederey commented on client/internal/ui/app.go:
```diff
@@ -0,0 +1,559 @@
+}
+
+// App represents the main application UI controller.
+type App struct {
```
И тут  App имеет слишком много ответственностей - раздели на отдельные компоненты: UIManager, EventHandler, DataManager

DanilNaum replied:
так там по идее все и так вынесено по максимуму все остальное непосредственно касается ui обработчики нажатия кнопок должны быть в пакете с ui поскольку - это методы *tview.Application, на все события в ui накинуты обработчики которые переносятся в другой пакет, а работа с данными используется только для отображения их в ui. просто пытаюсь понять, что там можно вынести и особо такого и не нахожу

Perederey commented on server/pkg/hasher/hasher.go:
```diff
@@ -0,0 +1,150 @@
+	params *Params
+}
+
+// NewArgon2Hasher создает новый Argon2Hasher
```
реализация Argon2 хэшера, не встречал такой еще!

DanilNaum replied:
не знаю я что-то нагуглил, вроде нормальный алгоритм, занимал место в самых надежных в 15 году, понятно, что возможно уже устарел, но чтоб прям что-то хорошее найти там отдельное исследование можно проводить, мне показалось, что для учебного проекта алгоритм шифорвания может быть любой и это не сильно принципиально

Perederey reviewed (commented):
Привет! Отлично справился с заданием 

Из того, на что стоит обратить внимание:
- Слишком много логики в main.go и UI компонентах - стоит вынести в отдельные сервисы
- Есть дублирование SQL кода и некоторые функции слишком длинные

А так молодец! Хорошая архитектура с интерфейсами, отличные тесты и правильное использование современных Go практик!

Commit:
refactor: split usecase interface into smaller interfaces and fix formatting

- Split monolithic usecase interface into recordCreator, recordReader, recordUpdater, and recordDeleter interfaces
- Extract downloadFile logic into separate method in UI forms
- Fix formatting and indentation issues in server main.go and mapper.go
- Add missing TextContent field mapping and FileData initialization in record mapper
- Add comment explaining async file content loading for file type records

Commit:
refactor: export struct types and reorganize main.go initialization

- Export grpcRecordHandler, grpcAuthHandler, grpcServer, storage, and config structs by capitalizing their names
- Refactor main.go to extract dependency initialization into separate functions for better organization and maintainability
- Add proper resource cleanup with defer pattern
- Remove unused app package initialization
